### ISSUE TITLE
New function that returns the file location

### ISSUE THREAD

**Issue by Robinlovelace** [ITSLeeds/osmextract#121]
E.g. called `oe_filelocate()`. Example showing what I'm trying to achieve - not sure why we rename the file with spaces, would it not be better as `isle-of-wight-latest.osm.pbf`, not least to be faithful to the original data source? Possibly not, but you can see it's hard to locate and that a function to find it would be useful:

``` r
remotes::install_github("itsleeds/osmextract")
#> Using github PAT from envvar GITHUB_PAT
#> Skipping install of 'osmextract' from a github remote, the SHA1 (dac00af6) has not changed since last install.
#>   Use `force = TRUE` to force installation
library(osmextract)
#> Data (c) OpenStreetMap contributors, ODbL 1.0. https://www.openstreetmap.org/copyright.
#> Any product made from OpenStreetMap must cite OSM as the data source.
#> Geofabrik data are taken from https://download.geofabrik.de/
#> For usage details of bbbike data see https://download.bbbike.org/osm/

oe_download("Isle of Wight")
#> Error in infer_provider_from_url(file_url): Cannot infer the provider from the url, please specify it.
oe_download("Isle of Wight", provider = "geofabrik")
#> Warning in utils::download.file(url = file_url, destfile = file_path, mode =
#> "wb", : URL 'http://Isle of Wight/': status was 'URL using bad/illegal format or
#> missing URL'
#> Error in utils::download.file(url = file_url, destfile = file_path, mode = "wb", : cannot open URL 'Isle of Wight'
iow_details = oe_match("Isle of Wight")
iow_details
#> $url
#> [1] "https://download.geofabrik.de/europe/great-britain/england/isle-of-wight-latest.osm.pbf"
#> 
#> $file_size
#> [1] 6877468
file_name = basename(iow_details$url)
file_location = file.path(oe_download_directory(), file_name)
file_location
#> [1] "/mnt/57982e2a-2874-4246-a6fe-115c199bc6bd/data/osm/isle-of-wight-latest.osm.pbf"
file.exists(file_location)
#> [1] FALSE
list.files(oe_download_directory(), pattern = "Isle")
#> [1] "Isle of Man.osm.pbf"   "Isle of Wight.osm.pbf"
```

<sup>Created on 2020-09-04 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by agila5** [comment]
I'm sorry but I'm not sure I understand the problem. 

> `oe_download("Isle of Wight")
`> #> Error in infer_provider_from_url(file_url): Cannot infer the provider from the url, please specify it.

I would say that this is not really a problem since the input of `oe_download()` should be the URL of the file that you want to download. Again, maybe we can improve the error message here, but I'm not sure how to check if the input parameter "looks like" a URL or the user passed the wrong argument (i.e. the name of the place instead of the URL). 

If you want to obtain the path of the `.pbf` or `.gpkg` file associated with a place, then you can run: 

``` r
# update package
remotes::install_github("itsleeds/osmextract")
#> Skipping install of 'osmextract' from a github remote, the SHA1 (dac00af6) has not changed since last install.
#>   Use `force = TRUE` to force installation
# load package
library(osmextract)
#> Data (c) OpenStreetMap contributors, ODbL 1.0. https://www.openstreetmap.org/copyright.
#> Any product made from OpenStreetMap must cite OSM as the data source.
#> Geofabrik data are taken from https://download.geofabrik.de/
#> For usage details of bbbike data see https://download.bbbike.org/osm/

oe_get("Isle of Wight", download_only = TRUE, skip_vectortranslate = TRUE)
#> [1] "C:\\Users\\Utente\\AppData\\Local\\Temp\\Rtmps3eVbL/geofabrik_isle-of-wight-latest.osm.pbf"
oe_get("Isle of Wight", download_only = TRUE)
#> [1] "C:\\Users\\Utente\\AppData\\Local\\Temp\\Rtmps3eVbL/geofabrik_isle-of-wight-latest.gpkg"
```

<sup>Created on 2020-09-04 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

Do you want to create a wrapper around this behaviour of `oe_get()`?

**Comment by Robinlovelace** [comment]
Yes and I'm giving it a bash now...

**Comment by Robinlovelace** [comment]
Since the commit above you can do:

``` r
remotes::install_github("itsleeds/osmextract", "oe_file")
#> Using github PAT from envvar GITHUB_PAT
#> Skipping install of 'osmextract' from a github remote, the SHA1 (63b02403) has not changed since last install.
#>   Use `force = TRUE` to force installation
library(osmextract)
#> Data (c) OpenStreetMap contributors, ODbL 1.0. https://www.openstreetmap.org/copyright.
#> Any product made from OpenStreetMap must cite OSM as the data source.
#> Geofabrik data are taken from https://download.geofabrik.de/
#> For usage details of bbbike data see https://download.bbbike.org/osm/

res = oe_get("ITS", provider = "test", max_string_dist = 9)
oe_file("its")
#> [1] "/mnt/57982e2a-2874-4246-a6fe-115c199bc6bd/data/osm/test_its-example.gpkg"   
#> [2] "/mnt/57982e2a-2874-4246-a6fe-115c199bc6bd/data/osm/test_its-example.osm.pbf"
oe_file("Isle of Wight")
#> [1] "/mnt/57982e2a-2874-4246-a6fe-115c199bc6bd/data/osm/Isle of Wight.osm.pbf"
oe_file("Namibia") # not downloaded yet
#> No file associated with that place name could be found.
oe_file("Non-existant-place")
#> No file associated with that place name could be found.

res = oe_get("ITS", provider = "test", max_string_dist = 9)
oe_get("its leeds", download_only = TRUE, skip_vectortranslate = TRUE)
#> Error: String distance between best match and the input place is 5, while the maximum threshold distance is equal to 1. You should increase the max_string_dist parameter, look for a closer match in the chosen provider database or consider using a different match_by variable.
oe_get("Isle of Wight", download_only = TRUE, skip_vectortranslate = TRUE)
#> [1] "/mnt/57982e2a-2874-4246-a6fe-115c199bc6bd/data/osm/geofabrik_isle-of-wight-latest.osm.pbf"
oe_get("Namibia", download_only = TRUE, skip_vectortranslate = TRUE) # not downloaded yet
#> [1] "/mnt/57982e2a-2874-4246-a6fe-115c199bc6bd/data/osm/geofabrik_namibia-latest.osm.pbf"
oe_get("Non-existant-place", download_only = TRUE, skip_vectortranslate = TRUE)
#> Error: String distance between best match and the input place is 11, while the maximum threshold distance is equal to 1. You should increase the max_string_dist parameter, look for a closer match in the chosen provider database or consider using a different match_by variable.
```

<sup>Created on 2020-09-06 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

Note the differences:

- `oe_get()` tries to download the file if it's not there already
- Less typing, more output (it shows all files)

Thoughts @agila5 ? I'm confident this will be a + for usability.

**Comment by Robinlovelace** [comment]
Closed in #123
